server/src/services/ConstructionService.ts

```diff
@@ -69,13 +71,27 @@ const constructionPreprocess = async ({ body: { operations } }: ApiRequest<Const
     throwError(ApiError.INVALID_OPERATIONS_LENGTH);
   }
 
-  const sender = new AccountIdentifier(
-    operations.find(({ amount: { value } }) => new BN(value).isNeg()).account.address,
-  );
+  const address: string = operations.find(({ amount: { value } }) => new BN(value).isNeg()).account.address;
 
-  const required_public_keys = [sender];
+  const isValidVaraAddress = checkAddress(address, 137)[0]
+  const isValidGearAddress = checkAddress(address, 42)[0]
```

Comment:
I used the prefixes from here:
https://github.com/gear-tech/gear/blob/dd05e9e12e3b36e727a8c619e859e755b6ad91cf/utils/gear-replay-cli/src/main.rs#L45-L46